src/dbnode/persist/fs/merger_test.go

```diff
@@ -463,8 +469,10 @@ func testMergeWith(
 		BlockStart: startTime,
 	}
 	mergeWith := mockMergeWithFromData(t, ctrl, diskData, mergeTargetData)
-	err := merger.Merge(fsID, mergeWith, 1, preparer, nsCtx, &persist.NoOpColdFlushNamespace{})
+	close, err := merger.Merge(fsID, mergeWith, 1, preparer, nsCtx, &persist.NoOpColdFlushNamespace{})
 	require.NoError(t, err)
+	require.NoError(t, close())
```

Comment:
nit: check `require.False(t, deferClosed)` before this to ensure we're not closing too early?

Reply:
+1